Tools/build/generate_global_objects.py

```diff
@@ -385,14 +385,22 @@ def generate_global_object_finalizers(immortal_objects):
         printer.write(before)
         printer.write(START)
         printer.write('#ifdef Py_DEBUG')
+
+        printer.write("""
+static inline void
+_PyStaticObject_CheckRefcnt(PyObject *obj) {
+    if (Py_REFCNT(obj) < _PyObject_IMMORTAL_REFCNT) {
+        _PyObject_ASSERT_FAILED_MSG(obj,
+            "immortal object has less refcnt than expected "
+            "_PyObject_IMMORTAL_REFCNT");
+    }
+}
```

Comment:
Rather that being part of the generated code, you could put this directly in pycore_global_objects_fini_generated.h, before the `/* The following is auto-generated by Tools/build/generate_global_objects.py. */` marker line.

Reply:
Aha, I didn't notice that only some parts of this file are generated. Ok, I added my new function at the top.